From b1cda4b0105cd2a1cb8a180aa1d51832cbe721c6 Mon Sep 17 00:00:00 2001
From: "yi.liu" <yi.liu@amlogic.com>
Date: Thu, 25 May 2023 11:09:00 +0800
Subject: [PATCH 7/8] amlbian: Optimize kernel configuration for debian [1/1]

PD#SWPL-125374

Problem:
Optimize kernel configuration for debian.

Solution:
Optimize kernel configuration for debian.

Verify:
local

Signed-off-by: yi.liu <yi.liu@amlogic.com>
Change-Id: Id666b52f536eb3d742060310f549bcdcbf1e418b
---
 conf/machine/mesong12b-w400-debian-5.15.conf  |   3 +-
 conf/machine/mesonsm1-ac200-debian-5.15.conf  |   3 +-
 conf/machine/mesont7c-an400-debian-5.15.conf  |   3 +-
 recipes-kernel/linux/5.15/amlbian.cfg         |  76 +++++++++++
 recipes-kernel/linux/5.15/g12b-debian.cfg     | 118 ------------------
 recipes-kernel/linux/5.15/sm1-debian.cfg      | 118 ------------------
 recipes-kernel/linux/5.15/t7-debian.cfg       |  49 --------
 recipes-kernel/linux/linux-meson_5.15.bb      |   6 +-
 .../media_modules/media-modules_5.15.bb       |  13 --
 .../aml-tools/aml-img-packer-native_1.0.bb    |   3 -
 10 files changed, 85 insertions(+), 307 deletions(-)
 create mode 100644 recipes-kernel/linux/5.15/amlbian.cfg
 delete mode 100644 recipes-kernel/linux/5.15/g12b-debian.cfg
 delete mode 100644 recipes-kernel/linux/5.15/sm1-debian.cfg
 delete mode 100644 recipes-kernel/linux/5.15/t7-debian.cfg

diff --git a/conf/machine/mesong12b-w400-debian-5.15.conf b/conf/machine/mesong12b-w400-debian-5.15.conf
index 7c178045..e521ff04 100644
--- a/conf/machine/mesong12b-w400-debian-5.15.conf
+++ b/conf/machine/mesong12b-w400-debian-5.15.conf
@@ -43,6 +43,7 @@ KERNEL_MODULE_AUTOLOAD += " hci_uart"
 KERNEL_MODULE_PROBECONF += "hci_uart"
 KERNEL_MODULE_AUTOLOAD += "meson-cpufreq"
 
+DISTRO_FEATURES_append = " amlbian"
 DISTRO_FEATURES_append = " drm"
 DISTRO_FEATURES_append = " gpt-partition"
 DISTRO_FEATURES_append = " kernel_515"
@@ -81,7 +82,7 @@ DISTRO_FEATURES_append = " gst-plugin-overlay"
 DISTRO_FEATURES_append = " gst-plugin-nn"
 DISTRO_FEATURES_append = " detect-library"
 DISTRO_FEATURES_append = " gst-plugin-conv"
-OVERRIDES .= ":w400:g12b-debian:onepass"
+OVERRIDES .= ":w400:g12b:onepass"
 MACHINEOVERRIDES .= ":client"
 
 EXTRA_BLUETOOTH_STUFF := "${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', bb.utils.contains('DISTRO_FEATURES', 'bluez5', ' bluez5', ' bluez4', d), '', d)}"
diff --git a/conf/machine/mesonsm1-ac200-debian-5.15.conf b/conf/machine/mesonsm1-ac200-debian-5.15.conf
index 9011e3fe..d3ca1c36 100644
--- a/conf/machine/mesonsm1-ac200-debian-5.15.conf
+++ b/conf/machine/mesonsm1-ac200-debian-5.15.conf
@@ -42,6 +42,7 @@ KERNEL_MODULE_AUTOLOAD += " hci_uart"
 KERNEL_MODULE_PROBECONF += "hci_uart"
 KERNEL_MODULE_AUTOLOAD += "meson-cpufreq"
 
+DISTRO_FEATURES_append = " amlbian"
 DISTRO_FEATURES_append = " drm"
 DISTRO_FEATURES_append = " gpt-partition"
 DISTRO_FEATURES_append = " kernel_515"
@@ -78,7 +79,7 @@ DISTRO_FEATURES_append = " gst-plugin-overlay"
 DISTRO_FEATURES_append = " gst-plugin-nn"
 DISTRO_FEATURES_append = " detect-library"
 DISTRO_FEATURES_append = " gst-plugin-conv"
-OVERRIDES .= ":ac200:sm1-debian:onepass"
+OVERRIDES .= ":ac200:sm1:onepass"
 MACHINEOVERRIDES .= ":client"
 
 EXTRA_BLUETOOTH_STUFF := "${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', bb.utils.contains('DISTRO_FEATURES', 'bluez5', ' bluez5', ' bluez4', d), '', d)}"
diff --git a/conf/machine/mesont7c-an400-debian-5.15.conf b/conf/machine/mesont7c-an400-debian-5.15.conf
index 64d14f63..1c42ed78 100644
--- a/conf/machine/mesont7c-an400-debian-5.15.conf
+++ b/conf/machine/mesont7c-an400-debian-5.15.conf
@@ -46,6 +46,7 @@ KERNEL_MODULE_AUTOLOAD += " hci_uart"
 KERNEL_MODULE_PROBECONF += "hci_uart"
 KERNEL_MODULE_AUTOLOAD += "meson-cpufreq"
 
+DISTRO_FEATURES_append = " amlbian"
 DISTRO_FEATURES_append = " drm"
 DISTRO_FEATURES_append = " gpt-partition"
 DISTRO_FEATURES_append = " kernel_515"
@@ -87,7 +88,7 @@ DISTRO_FEATURES_append = " gst-plugin-conv"
 DISTRO_FEATURES_append = " mediactrlsrc"
 
 
-OVERRIDES .= ":an400:t7-debian:onepass"
+OVERRIDES .= ":an400:t7:onepass"
 MACHINEOVERRIDES .= ":client"
 
 EXTRA_BLUETOOTH_STUFF := "${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', bb.utils.contains('DISTRO_FEATURES', 'bluez5', ' bluez5', ' bluez4', d), '', d)}"
diff --git a/recipes-kernel/linux/5.15/amlbian.cfg b/recipes-kernel/linux/5.15/amlbian.cfg
new file mode 100644
index 00000000..d52c3201
--- /dev/null
+++ b/recipes-kernel/linux/5.15/amlbian.cfg
@@ -0,0 +1,76 @@
+CONFIG_DRM_MESON_EMULATE_FBDEV=y
+# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
+CONFIG_CGROUP_PIDS=y
+CONFIG_CGROUP_RDMA=y
+CONFIG_CGROUP_DEVICE=y
+CONFIG_CGROUP_PERF=y
+CONFIG_NET_CLS_CGROUP=y
+CONFIG_CGROUP_NET_CLASSID=y
+CONFIG_USER_NS=y
+CONFIG_PID_NS=y
+
+#
+# support squashfs
+#
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_FILE_CACHE=y
+# CONFIG_SQUASHFS_FILE_DIRECT is not set
+CONFIG_SQUASHFS_DECOMP_SINGLE=y
+# CONFIG_SQUASHFS_DECOMP_MULTI is not set
+# CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU is not set
+CONFIG_SQUASHFS_XATTR=y
+CONFIG_SQUASHFS_ZLIB=y
+CONFIG_SQUASHFS_LZ4=y
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_XZ=y
+CONFIG_SQUASHFS_ZSTD=y
+CONFIG_SQUASHFS_4K_DEVBLK_SIZE=y
+CONFIG_SQUASHFS_EMBEDDED=y
+CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
+
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_TMPFS_XATTR=y
+CONFIG_CFS_BANDWIDTH=y
+CONFIG_RT_GROUP_SCHED=y
+#
+# add for docker check-config.sh
+#
+CONFIG_BRIDGE_NETFILTER=y
+CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=y
+CONFIG_NETFILTER_XT_MATCH_IPVS=y
+CONFIG_POSIX_MQUEUE=y
+
+CONFIG_BLK_DEV_THROTTLING=y
+CONFIG_IP_VS=y
+CONFIG_IP_VS_NFCT=y
+CONFIG_IP_VS_PROTO_TCP=y
+CONFIG_IP_VS_PROTO_UDP=y
+CONFIG_IP_VS_RR=y
+CONFIG_SECURITY_APPARMOR=y
+CONFIG_VXLAN=y
+CONFIG_BRIDGE_VLAN_FILTERING=y
+CONFIG_IPVLAN=y
+CONFIG_MACVLAN=y
+CONFIG_BTRFS_FS=y
+CONFIG_BTRFS_FS_POSIX_ACL=y
+
+# docker, cgroup
+CONFIG_CGROUPS=y
+CONFIG_BLK_CGROUPS=y
+# CONFIG_CGROUP_DEBUG_BLK_CGROUP is not set
+CONFIG_CGROUP_WRITEBACK=y
+CONFIG_CGROUP_SCHED=y
+CONFIG_CGROUP_FREEZER=y
+CONFIG_CGROUP_HUGETLB=y
+CONFIG_CGROUP_DEVICE=y
+CONFIG_CGROUP_CPUACCT=y
+CONFIG_CGROUP_BPRF=y
+# CONFIG_CGROUP_DEBUG is not set
+CONFIG_SOCK_CGROUP_DATA=y
+CONFIG_NETFILTER_XT_MATCH_CGROUP=m
+CONFIG_CGROUP_NET_PRIO=y
+# cgroup memory
+CONFIG_MEMCG=y
+CONFIG_MEMCG_SWAP=y
+CONFIG_MEMCG_SWAP_ENABLED=y
+CONFIG_MEMCG_KMEM=y
\ No newline at end of file
diff --git a/recipes-kernel/linux/5.15/g12b-debian.cfg b/recipes-kernel/linux/5.15/g12b-debian.cfg
deleted file mode 100644
index 5f40365a..00000000
--- a/recipes-kernel/linux/5.15/g12b-debian.cfg
+++ /dev/null
@@ -1,118 +0,0 @@
-CONFIG_AMLOGIC_SERIAL_MESON=y
-CONFIG_USERFAULTFD=y
-CONFIG_AUTOFS4_FS=y
-CONFIG_VT=y
-CONFIG_CONSOLE_TRANSLATIONS=y
-CONFIG_HW_CONSOLE=y
-CONFIG_FHANDLE=y
-CONFIG_SERIAL_DEV_BUS=y
-CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
-CONFIG_SYSVIPC=y
-CONFIG_BT_RFCOMM=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-# CONFIG_BT_HCIBTSDIO is not set
-CONFIG_DRM_MESON_EMULATE_FBDEV=y
-CONFIG_EXFAT_FS=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-CONFIG_CGROUP_PIDS=y
-CONFIG_CGROUP_RDMA=y
-CONFIG_CGROUP_DEVICE=y
-CONFIG_CGROUP_PERF=y
-CONFIG_NET_CLS_CGROUP=y
-CONFIG_CGROUP_NET_CLASSID=y
-CONFIG_USER_NS=y
-CONFIG_PID_NS=y
-
-#
-# support squashfs
-#
-CONFIG_SQUASHFS=y
-CONFIG_SQUASHFS_FILE_CACHE=y
-# CONFIG_SQUASHFS_FILE_DIRECT is not set
-CONFIG_SQUASHFS_DECOMP_SINGLE=y
-# CONFIG_SQUASHFS_DECOMP_MULTI is not set
-# CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU is not set
-CONFIG_SQUASHFS_XATTR=y
-CONFIG_SQUASHFS_ZLIB=y
-CONFIG_SQUASHFS_LZ4=y
-CONFIG_SQUASHFS_LZO=y
-CONFIG_SQUASHFS_XZ=y
-CONFIG_SQUASHFS_ZSTD=y
-CONFIG_SQUASHFS_4K_DEVBLK_SIZE=y
-CONFIG_SQUASHFS_EMBEDDED=y
-CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
-
-CONFIG_TMPFS_POSIX_ACL=y
-CONFIG_TMPFS_XATTR=y
-CONFIG_CFS_BANDWIDTH=y
-CONFIG_RT_GROUP_SCHED=y
-
-CONFIG_ARM64_ERRATUM_2441007=y
-CONFIG_REGULATOR_GPIO=m
-CONFIG_AMLOGIC_DRM_VPU=y
-CONFIG_AMLOGIC_DRM_HDMI=y
-CONFIG_AMLOGIC_DRM_USE_ION=y
-CONFIG_AMLOGIC_DRM_EMULATE_FBDEV=y
-CONFIG_AMLOGIC_FIXED_AREA_CMA=m
-CONFIG_AMLOGIC_PCIE_DMA_OPS=y
-CONFIG_AMLOGIC_FAKE_SMMU=y
-CONFIG_AMLOGIC_DMC_DEV_ACCESS=y
-CONFIG_AMLOGIC_HWSPINLOCK=m
-CONFIG_AMLOGIC_ALGORITHM=m
-CONFIG_AMLOGIC_SND_SOC_AD82128=m
-CONFIG_AMLOGIC_SND_SOC_TAS5805=m
-CONFIG_INIT_ON_ALLOC_DEFAULT_ON=y
-CONFIG_AS_HAS_NON_CONST_LEB128=y
-CONFIG_AMLOGIC_MEDIA_CAMERA=m
-
-#Remove some configs
-# CONFIG_AMLOGIC_PINCTRL_MESON_S4 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_SC2 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_T3 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_T7 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_S4 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_SC2 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_T3 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_T7 is not set
-CONFIG_KALLSYMS_ALL=n
-CONFIG_KASAN=n
-CONFIG_KFENCE=n
-CONFIG_F2FS_FS=n
-CONFIG_QUOTA=n
-CONFIG_VIRTUALIZATION=n
-CONFIG_IPV6=m
-CONFIG_FB=n
-CONFIG_CAN=n
-CONFIG_USB_NET_DRIVERS=n
-CONFIG_SERIAL_8250=n
-CONFIG_SERIAL_AMBA_PL011=n
-CONFIG_SERIAL_SAMSUNG=n
-CONFIG_SERIAL_QCOM_GENI=n
-CONFIG_SERIAL_SPRD=n
-CONFIG_MII=m
-CONFIG_INCREMENTAL_FS=n
-CONFIG_CRYPTO_ZSTD=n
-CONFIG_CRYPTO_LZO=n
-CONFIG_ZSTD_COMPRESS=n
-CONFIG_RD_ZSTD=n
-CONFIG_RD_LZ4=n
-CONFIG_DECOMPRESS_ZSTD=n
-CONFIG_ELF_CORE=n
-CONFIG_SCSI_UFSHCD=n
-CONFIG_PERF_EVENTS=n
-CONFIG_SLUB_DEBUG=n
-CONFIG_BT_DEBUGFS=n
-CONFIG_BUG_ON_DATA_CORRUPTION=n
-CONFIG_DEBUG_LIST=n
-CONFIG_BLK_DEBUG_FS=n
-CONFIG_CMA_DEBUGFS=n
-CONFIG_DEBUG_KINFO=n
-CONFIG_ANDROID_DEBUG_SYMBOLS=n
-CONFIG_DYNAMIC_DEBUG_CORE=n
-CONFIG_DEBUG_BUGVERBOSE=n
-CONFIG_DEBUG_INFO=n
-CONFIG_DEBUG_INFO_DWARF4=n
-CONFIG_DEBUG_INFO_BTF=n
-CONFIG_DEBUG_INFO_BTF_MODULES=n
-CONFIG_DEBUG_MISC=n
-CONFIG_DEBUG_STACK_USAGE=n
diff --git a/recipes-kernel/linux/5.15/sm1-debian.cfg b/recipes-kernel/linux/5.15/sm1-debian.cfg
deleted file mode 100644
index 5f40365a..00000000
--- a/recipes-kernel/linux/5.15/sm1-debian.cfg
+++ /dev/null
@@ -1,118 +0,0 @@
-CONFIG_AMLOGIC_SERIAL_MESON=y
-CONFIG_USERFAULTFD=y
-CONFIG_AUTOFS4_FS=y
-CONFIG_VT=y
-CONFIG_CONSOLE_TRANSLATIONS=y
-CONFIG_HW_CONSOLE=y
-CONFIG_FHANDLE=y
-CONFIG_SERIAL_DEV_BUS=y
-CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
-CONFIG_SYSVIPC=y
-CONFIG_BT_RFCOMM=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-# CONFIG_BT_HCIBTSDIO is not set
-CONFIG_DRM_MESON_EMULATE_FBDEV=y
-CONFIG_EXFAT_FS=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-CONFIG_CGROUP_PIDS=y
-CONFIG_CGROUP_RDMA=y
-CONFIG_CGROUP_DEVICE=y
-CONFIG_CGROUP_PERF=y
-CONFIG_NET_CLS_CGROUP=y
-CONFIG_CGROUP_NET_CLASSID=y
-CONFIG_USER_NS=y
-CONFIG_PID_NS=y
-
-#
-# support squashfs
-#
-CONFIG_SQUASHFS=y
-CONFIG_SQUASHFS_FILE_CACHE=y
-# CONFIG_SQUASHFS_FILE_DIRECT is not set
-CONFIG_SQUASHFS_DECOMP_SINGLE=y
-# CONFIG_SQUASHFS_DECOMP_MULTI is not set
-# CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU is not set
-CONFIG_SQUASHFS_XATTR=y
-CONFIG_SQUASHFS_ZLIB=y
-CONFIG_SQUASHFS_LZ4=y
-CONFIG_SQUASHFS_LZO=y
-CONFIG_SQUASHFS_XZ=y
-CONFIG_SQUASHFS_ZSTD=y
-CONFIG_SQUASHFS_4K_DEVBLK_SIZE=y
-CONFIG_SQUASHFS_EMBEDDED=y
-CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
-
-CONFIG_TMPFS_POSIX_ACL=y
-CONFIG_TMPFS_XATTR=y
-CONFIG_CFS_BANDWIDTH=y
-CONFIG_RT_GROUP_SCHED=y
-
-CONFIG_ARM64_ERRATUM_2441007=y
-CONFIG_REGULATOR_GPIO=m
-CONFIG_AMLOGIC_DRM_VPU=y
-CONFIG_AMLOGIC_DRM_HDMI=y
-CONFIG_AMLOGIC_DRM_USE_ION=y
-CONFIG_AMLOGIC_DRM_EMULATE_FBDEV=y
-CONFIG_AMLOGIC_FIXED_AREA_CMA=m
-CONFIG_AMLOGIC_PCIE_DMA_OPS=y
-CONFIG_AMLOGIC_FAKE_SMMU=y
-CONFIG_AMLOGIC_DMC_DEV_ACCESS=y
-CONFIG_AMLOGIC_HWSPINLOCK=m
-CONFIG_AMLOGIC_ALGORITHM=m
-CONFIG_AMLOGIC_SND_SOC_AD82128=m
-CONFIG_AMLOGIC_SND_SOC_TAS5805=m
-CONFIG_INIT_ON_ALLOC_DEFAULT_ON=y
-CONFIG_AS_HAS_NON_CONST_LEB128=y
-CONFIG_AMLOGIC_MEDIA_CAMERA=m
-
-#Remove some configs
-# CONFIG_AMLOGIC_PINCTRL_MESON_S4 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_SC2 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_T3 is not set
-# CONFIG_AMLOGIC_PINCTRL_MESON_T7 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_S4 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_SC2 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_T3 is not set
-# CONFIG_AMLOGIC_COMMON_CLK_T7 is not set
-CONFIG_KALLSYMS_ALL=n
-CONFIG_KASAN=n
-CONFIG_KFENCE=n
-CONFIG_F2FS_FS=n
-CONFIG_QUOTA=n
-CONFIG_VIRTUALIZATION=n
-CONFIG_IPV6=m
-CONFIG_FB=n
-CONFIG_CAN=n
-CONFIG_USB_NET_DRIVERS=n
-CONFIG_SERIAL_8250=n
-CONFIG_SERIAL_AMBA_PL011=n
-CONFIG_SERIAL_SAMSUNG=n
-CONFIG_SERIAL_QCOM_GENI=n
-CONFIG_SERIAL_SPRD=n
-CONFIG_MII=m
-CONFIG_INCREMENTAL_FS=n
-CONFIG_CRYPTO_ZSTD=n
-CONFIG_CRYPTO_LZO=n
-CONFIG_ZSTD_COMPRESS=n
-CONFIG_RD_ZSTD=n
-CONFIG_RD_LZ4=n
-CONFIG_DECOMPRESS_ZSTD=n
-CONFIG_ELF_CORE=n
-CONFIG_SCSI_UFSHCD=n
-CONFIG_PERF_EVENTS=n
-CONFIG_SLUB_DEBUG=n
-CONFIG_BT_DEBUGFS=n
-CONFIG_BUG_ON_DATA_CORRUPTION=n
-CONFIG_DEBUG_LIST=n
-CONFIG_BLK_DEBUG_FS=n
-CONFIG_CMA_DEBUGFS=n
-CONFIG_DEBUG_KINFO=n
-CONFIG_ANDROID_DEBUG_SYMBOLS=n
-CONFIG_DYNAMIC_DEBUG_CORE=n
-CONFIG_DEBUG_BUGVERBOSE=n
-CONFIG_DEBUG_INFO=n
-CONFIG_DEBUG_INFO_DWARF4=n
-CONFIG_DEBUG_INFO_BTF=n
-CONFIG_DEBUG_INFO_BTF_MODULES=n
-CONFIG_DEBUG_MISC=n
-CONFIG_DEBUG_STACK_USAGE=n
diff --git a/recipes-kernel/linux/5.15/t7-debian.cfg b/recipes-kernel/linux/5.15/t7-debian.cfg
deleted file mode 100644
index 96942e2d..00000000
--- a/recipes-kernel/linux/5.15/t7-debian.cfg
+++ /dev/null
@@ -1,49 +0,0 @@
-CONFIG_USERFAULTFD=y
-CONFIG_AUTOFS4_FS=y
-CONFIG_VT=y
-CONFIG_CONSOLE_TRANSLATIONS=y
-CONFIG_HW_CONSOLE=y
-CONFIG_FHANDLE=y
-CONFIG_ANDROID_LOGGER=y
-CONFIG_SERIAL_DEV_BUS=y
-CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
-CONFIG_SYSVIPC=y
-CONFIG_BT_RFCOMM=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-# CONFIG_BT_HCIBTSDIO is not set
-CONFIG_DRM_MESON_EMULATE_FBDEV=y
-CONFIG_EXFAT_FS=y
-# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
-CONFIG_CGROUP_PIDS=y
-CONFIG_CGROUP_RDMA=y
-CONFIG_CGROUP_DEVICE=y
-CONFIG_CGROUP_PERF=y
-CONFIG_NET_CLS_CGROUP=y
-CONFIG_CGROUP_NET_CLASSID=y
-CONFIG_USER_NS=y
-CONFIG_PID_NS=y
-
-#
-# support squashfs
-#
-CONFIG_SQUASHFS=y
-CONFIG_SQUASHFS_FILE_CACHE=y
-# CONFIG_SQUASHFS_FILE_DIRECT is not set
-CONFIG_SQUASHFS_DECOMP_SINGLE=y
-# CONFIG_SQUASHFS_DECOMP_MULTI is not set
-# CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU is not set
-CONFIG_SQUASHFS_XATTR=y
-CONFIG_SQUASHFS_ZLIB=y
-CONFIG_SQUASHFS_LZ4=y
-CONFIG_SQUASHFS_LZO=y
-CONFIG_SQUASHFS_XZ=y
-CONFIG_SQUASHFS_ZSTD=y
-CONFIG_SQUASHFS_4K_DEVBLK_SIZE=y
-CONFIG_SQUASHFS_EMBEDDED=y
-CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
-
-CONFIG_TMPFS_POSIX_ACL=y
-CONFIG_TMPFS_XATTR=y
-CONFIG_CFS_BANDWIDTH=y
-CONFIG_RT_GROUP_SCHED=y
-
diff --git a/recipes-kernel/linux/linux-meson_5.15.bb b/recipes-kernel/linux/linux-meson_5.15.bb
index 0b6db189..920ef333 100644
--- a/recipes-kernel/linux/linux-meson_5.15.bb
+++ b/recipes-kernel/linux/linux-meson_5.15.bb
@@ -16,9 +16,6 @@ SRC_URI_append_s4 = " file://s4.cfg"
 SRC_URI_append_t7 = " file://t7.cfg"
 SRC_URI_append_g12b = " file://g12b.cfg"
 SRC_URI_append_sm1 = " file://sm1.cfg"
-SRC_URI_append_t7-debian = " file://t7-debian.cfg"
-SRC_URI_append_g12b-debian = " file://g12b-debian.cfg"
-SRC_URI_append_sm1-debian = " file://sm1-debian.cfg"
 
 SRC_URI += "file://common.cfg"
 
@@ -28,6 +25,9 @@ SRC_URI += "${@bb.utils.contains('DISTRO_FEATURES', 'nand', 'file://nand.cfg', '
 # Enable selinux support in the kernel if the feature is enabled
 SRC_URI += "${@bb.utils.contains('DISTRO_FEATURES', 'selinux', 'file://selinux.cfg', '', d)}"
 
+# Enable debian support in the kernel if the feature is enabled
+SRC_URI_append = "${@bb.utils.contains('DISTRO_FEATURES', 'amlbian', ' file://amlbian.cfg', '', d)}"
+
 #For common patches
 KDIR = "aml-5.15"
 SRC_URI_append = " ${@get_patch_list_with_path('${AML_PATCH_PATH}/kernel/${KDIR}')}"
diff --git a/recipes-kernel/media_modules/media-modules_5.15.bb b/recipes-kernel/media_modules/media-modules_5.15.bb
index 354a7d7f..7ff9dbb9 100644
--- a/recipes-kernel/media_modules/media-modules_5.15.bb
+++ b/recipes-kernel/media_modules/media-modules_5.15.bb
@@ -72,10 +72,6 @@ MEDIA_CONFIGS_append_t7 = "\
                  CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
                  CONFIG_AMLOGIC_MEDIA_VENC_MULTI=m \
                  "
-MEDIA_CONFIGS_append_t7-debian = "\
-                 CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
-                 CONFIG_AMLOGIC_MEDIA_VENC_MULTI=m \
-                 "
 
 MEDIA_CONFIGS_append_sc2 = "\
                  CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
@@ -86,12 +82,6 @@ MEDIA_CONFIGS_append_g12b = "\
 MEDIA_CONFIGS_append_sm1 = "\
                  CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
                  "
-MEDIA_CONFIGS_append_g12b-debian = "\
-                 CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
-                 "
-MEDIA_CONFIGS_append_sm1-debian = "\
-                 CONFIG_AMLOGIC_MEDIA_VENC_JPEG=m \
-                 "
 
 S = "${WORKDIR}/git"
 EXTRA_OEMAKE='-C ${STAGING_KERNEL_DIR} M="${S}/drivers" EXTRA_CFLAGS=${MEDIA_CONFIGS} modules'
@@ -131,12 +121,9 @@ KERNEL_MODULE_AUTOLOAD += "amvdec_ports"
 
 KERNEL_MODULE_AUTOLOAD_append_t3 = " encoder amlogic-jpegenc amlogic-multienc"
 KERNEL_MODULE_AUTOLOAD_append_t7 = " encoder amlogic-jpegenc amlogic-multienc"
-KERNEL_MODULE_AUTOLOAD_append_t7-debian = " amlogic-jpegenc amlogic-multienc"
 KERNEL_MODULE_AUTOLOAD_append_sc2 = " encoder jpegenc vpu"
 KERNEL_MODULE_AUTOLOAD_append_g12b = " amlogic-encoder amlogic-jpegenc amlogic-vpu"
-KERNEL_MODULE_AUTOLOAD_append_g12b-debian = " amlogic-encoder amlogic-jpegenc amlogic-vpu"
 KERNEL_MODULE_AUTOLOAD_append_sm1 = " amlogic-encoder amlogic-jpegenc amlogic-vpu"
-KERNEL_MODULE_AUTOLOAD_append_sm1-debian = " amlogic-encoder amlogic-jpegenc amlogic-vpu"
 
 KERNEL_MODULE_PROBECONF += "amvdec_ports amvdec_mh264"
 module_conf_amvdec_ports = "options amvdec_ports multiplanar=1 vp9_need_prefix=1 av1_need_prefix=1"
diff --git a/recipes-support/aml-tools/aml-img-packer-native_1.0.bb b/recipes-support/aml-tools/aml-img-packer-native_1.0.bb
index efc609b4..aeee42d7 100644
--- a/recipes-support/aml-tools/aml-img-packer-native_1.0.bb
+++ b/recipes-support/aml-tools/aml-img-packer-native_1.0.bb
@@ -14,11 +14,9 @@ SOC_FAMILY_axg = "axg"
 SOC_FAMILY_txlx = "txlx"
 SOC_FAMILY_g12a = "g12a"
 SOC_FAMILY_g12b = "g12b"
-SOC_FAMILY_g12b-debian = "g12b"
 SOC_FAMILY_tm2 = "tm2"
 SOC_FAMILY_sc2 = "sc2"
 SOC_FAMILY_t7 = "t7"
-SOC_FAMILY_t7-debian = "t7"
 SOC_FAMILY_p1 = "p1"
 SOC_FAMILY_s4 = "s4"
 SOC_FAMILY_t5d = "t5d"
@@ -27,7 +25,6 @@ SOC_FAMILY_t3 = "t3"
 SOC_FAMILY_a213y = "a213y"
 SOC_FAMILY_s5 = "s5"
 SOC_FAMILY_sm1 = "g12a"
-SOC_FAMILY_sm1-debian = "g12a"
 
 SOC_BOARD = "default"
 SOC_BOARD_ab301 = "ab301"
-- 
2.25.1

